### PR TITLE
fix(nested rows): fixed border color and align

### PR DESCRIPTION
**Pull request by jagathgj**
Contributes to #2505 

Design review fixes done for the nested rows. Alignment of columns fixed as well as border color changed

#### What did you change?
scss changed and also condition for getCellProps and geRowStyles in useNestedRows changed

#### How did you test and verify your work?
storybook

### CODE REVIEW

**Comment by davidmenendez** [comment]
tests are failing

**Review by davidmenendez — DISMISSED** [review]

File: packages/cloud-cognitive/src/components/Datagrid/useNestedRows.js [hunk 1]
```diff
@@ -32,12 +32,10 @@ const useNestedRows = (hooks) => {
       style: {
         marginLeft: `${row.depth > 0 ? marginLeft : 0}px`,
```
Comment (elycheea) [inline packages/cloud-cognitive/src/components/Datagrid/useNestedRows.js:33]: It looks like we need to remove this margin too in v11?

Seeing extra spacing and text misalignment here.
image

File: packages/cloud-cognitive/src/components/Datagrid/styles/_useNestedRows.scss [hunk 2]
```diff
@@ -50,3 +50,21 @@
...
+.#{$block-class} tr.#{$block-class}__carbon-nested-row {
+  td:first-child:empty,
+  .#{$block-class}__expandable-row-cell {
+    border-bottom: none;
+  }
+}
+
+.#{c4p-settings.$carbon-prefix}--data-table
+  td.#{$block-class}__expandable-row-cell {
+  padding-left: 0;
```
Comment (elycheea) [inline packages/cloud-cognitive/src/components/Datagrid/styles/_useNestedRows.scss:63]: I think this should actually use `$spacing-03`.

Using `padding-left: 0`, the button gets shifted a little to the left.

Collapsed | Expanded
-----------|----------
image | image

**Review by elycheea — CHANGES_REQUESTED** [review]
This one we also need to remove the extra left margin plus adjust the spacing.
I do have this in a PR already but wasn’t able to push to your branch!

File: packages/cloud-cognitive/src/components/Datagrid/useNestedRows.js (hunk 1 above)
Comment (jagathgj) [reply packages/cloud-cognitive/src/components/Datagrid/useNestedRows.js]: Margin removed.

**Review by elycheea — COMMENTED** [review]
Looks good but just realized these should probably work onFocus as well.

image

Probably only when the arrow is focused so we may need to add some fixes to  `useNestedRowExpander.js` as well. (Applies to both v10 and v11 PRs.)

**Pull request merged**